File: rule-indexer-python/indexing_rules/capitalized_words_rule.py
from .base import IndexingRule
from logics.capitalized_word_count_logic import CapitalizedWordCountLogic

class CapitalizedWordsRule(IndexingRule):
    def __init__(self):
        self.logic = CapitalizedWordCountLogic()
    
    def apply_rule(self, content):
        count = self.logic.apply(content)
        print(f"Number of capitalized words: {count}")

File: rule-indexer-go/src/main/rules/capitalized_words_rule.go
package rules

import (
    "fmt"
    "rule-indexer-go/src/main/logics"
)

type CapitalizedWordsRule struct {
    logic *logics.CapitalizedWordCountLogic
}

func NewCapitalizedWordsRule() *CapitalizedWordsRule {
    return &CapitalizedWordsRule{
        logic: logics.NewCapitalizedWordCountLogic(),
    }
}

func (c *CapitalizedWordsRule) ApplyRule(content string) {
    count := c.logic.Apply(content)
    fmt.Printf("Number of capitalized words: %d\n", count)
}
